src/Test/Perf/infra/fetch_build.csx

```diff
@@ -4,39 +4,130 @@
 {
-    throw new InvalidOperationException($"Could not locate build with passing tests at location \"{sourceFolder}\".");
-}
+    var processNamesToKill = new[] {"csc", "vbc", "vbcscompiler", "devenv", "cpc" };
+    System.Diagnostics.Process.GetProcesses()
+        .Where(x => processNamesToKill.Any(p => x.ProcessName.ToLower().StartsWith(p)))
+        .ToList()
+        .ForEach(x => x.Kill());
 
-var latestBuildFolder = Path.Combine(sourceFolder, latestBuild);
+    //Cleanup the registry entries by CPC to make it ready for the next run 
+    var processInfo = new ProcessStartInfo(Environment.ExpandEnvironmentVariables(@"%SYSTEMDRIVE%\CPC\CPC.exe"));
+    processInfo.Arguments = @"/DisableArchive /Cleanup";
+    var process = Process.Start(processInfo);
 
-if (Directory.Exists(destinationFolder))
-{
-    Directory.Delete(destinationFolder, recursive: true);
+    // Shouldn't take more than 1 minute to cleanup
+    if(!process.WaitForExit(60000))
+    {
+        // If the CPC doesn't cleanup in a minute then we need to stop the automation and check the machine
+        var logFilePath = Path.Combine((Environment.ExpandEnvironmentVariables("%USERPROFILE%")), "PerfRunErrorLog.txt");
+        File.AppendAllLines(logFilePath, new[] { DateTime.Now.ToString()});
+        File.AppendAllLines(logFilePath, new[] { "CPC did not get cleaned up within 1 minute. Hence shutting down automation" });
+
+        Process.GetCurrentProcess().Kill();
+    }
 }
 
-CopyDirectory(latestBuildFolder, destinationFolder);
+void FindBuildAndWaitForTestToComplete(
+    string sourceFolder,
+    string destinationFolder,
+    string lastBuild,
+    string perfRunStatusPath)
+{
+    var lastBuildVersion = lastBuild == null ? null : Version.Parse(lastBuild);
+    foreach (var folder in Directory.GetDirectories(sourceFolder, "????????.?").Reverse())
+    {
+        var buildDirectoryName = Path.GetFileName(folder);
+        var currentBuildVersion = Version.Parse(buildDirectoryName);
+
+        // if we are no longer looking at the newer builds, then quit searching
+        if (lastBuildVersion != null && currentBuildVersion <= lastBuildVersion)
+        {
+            break;
+        }
+
+        if (SanityTestPassesForBuild(folder))
+        {
+            // Copy the binaries from the drop to Release binaries
+            var latestBuildFolder = Path.Combine(sourceFolder, folder);
+
+            if (Directory.Exists(destinationFolder))
+            {
+                Directory.Delete(destinationFolder, recursive: true);
+            }
+
+            CopyDirectory(latestBuildFolder, destinationFolder);
+
+            // The test files have been copied. Now is a good time to record the status for this build
+            File.WriteAllText(perfRunStatusPath, buildDirectoryName);
+
+            Console.WriteLine($"Starting tests for {latestBuildFolder}");
+            // Start Automation
+            var processInfo = new ProcessStartInfo(@"TriggerAutomation.bat");
```

Comment:
At this point, TriggerAutomation.bat seems like an unnecessary indirection.  Can we just rename  fetch_build.csx to automation.csx (and delete TriggerAutomation.bat/automation.csx)?